Controller/Checkout/SecondChance.php

```diff
@@ -141,102 +71,44 @@ class SecondChance extends \Magento\Framework\App\Action\Action
         \Magento\Customer\Model\CustomerFactory $customerFactory,
         \Magento\Customer\Model\SessionFactory $sessionFactory,
         \Magento\Customer\Api\CustomerRepositoryInterface $customerRepositoryInterface,
-        \Magento\Quote\Api\CartManagementInterface $quoteManagement,
         \Magento\Sales\Model\OrderIncrementIdChecker $orderIncrementIdChecker = null
-
     ) {
         parent::__construct($context);
-        $this->helper             = $helper;
-        $this->cart               = $cart;
-        $this->order              = $order;
-        $this->quote              = $quote;
-        $this->quoteRepository    = $quoteRepository;
-        $this->quoteFactory       = $quoteFactory;
-        $this->transaction        = $transaction;
-        $this->logger             = $logger;
-        $this->orderSender        = $orderSender;
-        $this->orderStatusFactory = $orderStatusFactory;
-        $this->quoteRecreate      = $quoteRecreate;
-
-        $this->accountConfig = $configProviderFactory->get('account');
-
-        $this->_curlClient            = $curl;
-        $this->_configProviderAccount = $configProviderAccount;
-        $this->_storeManager          = $storeManager;
-        $this->checkoutSession        = $checkoutSession;
-        $this->_encryptor             = $encryptor;
-        $this->priceCurrency          = $priceCurrency;
-        $this->jsonHelper             = $jsonHelper;
-        $this->jsonResultFactory      = $jsonResultFactory;
-        $this->messageManager         = $messageManager;
-        $this->_orderFactory          = $orderFactory;
-        $this->sequenceManager        = $sequenceManager;
-        $this->eavConfig              = $eavConfig;
-        $this->urlBuilder             = $urlBuilder;
-
-        $this->productFactory  = $productFactory;
-        $this->quoteManagement = $quoteManagement;
-        $this->customerSession = $customerSession;
-
-        $this->_customerFactory             = $customerFactory;
-        $this->_sessionFactory              = $sessionFactory;
-        $this->_customerRepositoryInterface = $customerRepositoryInterface;
-
-        $this->guestCart           = $guestCart;
-        $this->secondChanceFactory = $secondChanceFactory;
 
+        $this->logger                  = $logger;
+        $this->quoteRecreate           = $quoteRecreate;
+        $this->configProviderAccount   = $configProviderAccount;
+        $this->storeManager            = $storeManager;
+        $this->checkoutSession         = $checkoutSession;
+        $this->orderFactory            = $orderFactory;
+        $this->customerSession         = $customerSession;
+        $this->sessionFactory          = $sessionFactory;
+        $this->customerFactory         = $customerFactory;
+        $this->secondChanceFactory     = $secondChanceFactory;
         $this->orderIncrementIdChecker = $orderIncrementIdChecker ?: ObjectManager::getInstance()
```

Comment:
maybe we can stop using object manager ...

Reply:
Magento 2.4 core still used Object Manager https://github.com/magento/magento2/blob/2.4/app/code/Magento/Quote/Model/Quote.php#L492 
so for correct work Quote used same way